lambda/src/main/scala/pricemigrationengine/handlers/NotificationHandler.scala

```diff
@@ -32,27 +37,57 @@ object NotificationHandler extends CohortHandler {
     } yield HandlerOutput(isComplete = true)
   }
 
+  def itemHasEnoughNotificationPadding(today: LocalDate, cohortItem: CohortItem): Boolean = {
+    // To help with backward compatibility with existing tests, we apply this condition from 1st Dec 2022.
+    if (today.isBefore(LocalDate.of(2022, 12, 1))) {
+      true
+    } else {
+      cohortItem.startDate match {
+        case Some(sd) => today.plusDays(NotificationPaddingTimeDays).isBefore(sd)
+        case _        => false
+      }
+    }
+  }
+
   def sendNotification(brazeCampaignName: String)(
       cohortItem: CohortItem
   ): ZIO[EmailSender with SalesforceClient with CohortTable with Logging, Failure, Int] = {
-    val result = for {
-      sfSubscription <-
-        SalesforceClient
-          .getSubscriptionByName(cohortItem.subscriptionName)
-      count <-
-        if (sfSubscription.Status__c != Cancelled_Status) {
-          sendNotification(brazeCampaignName, cohortItem, sfSubscription)
-        } else {
-          putSubIntoCancelledStatus(cohortItem.subscriptionName)
-        }
-    } yield count
-
-    result.catchAll { failure =>
-      for {
-        _ <- Logging.error(
-          s"Subscription ${cohortItem.subscriptionName}: Failed to send price rise notification: $failure"
+
+    // We are starting with a simple check. That the item's startDate is at least NotificationPaddingTimeDays days away
+    // from the current day. This will avoid headaches caused by letters not being sent early enough relatively to
+    // previously computed start dats, and will detect any such problem when they happen and that
+    // before the letters are sent.
+
+    if (itemHasEnoughNotificationPadding(LocalDate.now(), cohortItem)) {
```

Comment:
Should avoid relative dates

Reply:
@kelvin-chappell  by relative dates do you mean avoiding using `LocalDate.now()` and using Zio's clock instead or avoiding relative dates altogether?

 @shtukas   even though using `LocalDate.now()` works well for this function and is easily testable, it's best to use Zio clock incase we do integration tests

Reply:
Yes sorry I don't mean avoid relative dates altogether.  I just mean use zio.Clock to get them lazily instead.

Reply:
Corrected. Thanks!